rmf_traffic_ros2/test/mock_monitor_nodes/missing_query_monitor.cpp

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <chrono>
+#include <thread>
+
+#include <rmf_traffic_ros2/schedule/internal_MonitorNode.hpp>
+#include <rmf_traffic/schedule/Database.hpp>
+
+#include <rclcpp/rclcpp.hpp>
+
+using namespace std::chrono_literals;
+
+class MissingQueryMonitorNode : public rmf_traffic_ros2::schedule::MonitorNode
+{
+public:
+  MissingQueryMonitorNode(
+    std::function<void(std::shared_ptr<rclcpp::Node>)> callback,
+    const rclcpp::NodeOptions& options)
+    : MonitorNode(callback, options, MonitorNode::no_automatic_setup)
+  {}
+
+  void announce_fail_over()
+  {
+    if (rclcpp::ok())
+    {
+      // Wait a while before announcing. This will ensure that mirrors pick up
+      // on the missing query via the query check logic, not the failover
+      // handling logic. (But the announcement still has to be made to ensure
+      // mirrors reconnect services.)
+      std::this_thread::sleep_for(5s);
+      RCLCPP_INFO(get_logger(), "Announcing fail over");
+      auto message = FailOverEvent();
```

Comment:
I wonder if the fail over event message should include the new schedule node edition value?

Reply:
Addressed by #87